docs/releaseNotes.md

```diff
@@ -0,0 +1,87 @@
+---
+title: Apache DistributedLog 0.5.0 Release Notes
+---
+
+This is the second Apache release of DistributedLog and the first release as a sub-project of Apache BookKeeper!
+
+The 0.5.0 release upgrades BookKeeper from Twitter `4.3.7-TWTTR-OSS` version to the official Apache `4.5.0` version.
+It is a big milestone in Apache BookKeeper & DistributedLog community, converaging the developement efforts for
+both communities and moving the distributedlog developement on the official Apache bookkeeper release.
+
+Apache DistributedLog users are encouraged to upgrade to 0.5.0. The technical details of this release are summarized
+below.
+
+## Highlights
+
+### BookKeeper Verion Upgrade
+
+The main change in 0.5.0 is bookkeeper version upgrade. We upgraded the bookkeeper version from Twitter's `4.3.7-TWTTR-OSS`
+to the official Apache `4.5.0` version. This upgrade brings in bookkeeper security feature and performance improvement (such as netty 4 upgrade).
+
+### Change Twitter Future to Java Future
+
+Prior to 0.5.0, the distributedlog API heavily depends on Twitter Future library and other scala dependency. In 0.5.0, we change
+the twitter future API to a separate module and replace the core API with Java8 CompletableFuture. It reduces the dependencies introduced
+by Scala and also avoid maintaining multiple versions for different scala versions.
+
+### Better buffer/memory management
+
+After upgrading to BookKeeper 4.5.0, we update the buffer and memory management in distributedlog to leverage netty's buffer management.
+It reduces object allocation, memory copies and improves the latency because less jvm garbages are produced.
```

Comment:
we probably to attach a benchmark result here.

Reply:
Hi @sijie. Is there any result available?

Reply:
Hi @sijie. Is there any result available?